### PR TITLE
Add support for ES2015 classes and extension

### LINKED ISSUE #2520: ES6: support class extends
Given

```
class Base {
   say() { .. }
}

class User extends Base { }
Polymer(User)
```

It's not working. tested on `v1.1.4`

possible related to https://github.com/Polymer/polymer/pull/2365#discussion_r37827984

### PR DESCRIPTION
**Pull request by azakus**
Supported Use Cases:

1. `class MyElement`
2. `class MyElement extends Polymer.Element`
3. `class MyElement extends Polymer.Extend('input')`
4. `class MyElement extends OtherPolymerElement`
5. `class MyElement extends Polymer.Extend('other-polymer-element')`

Fixes #2520

### CODE REVIEW

File: src/micro/extends.html [hunk 1]
```diff
@@ -106,4 +114,28 @@
 
   });
 
+  /**
+   * Utility function for extending native elements and Polymer elements with
+   * ES2015 class syntax.
+   *
+   * `Polymer.Extend()` will
```
Comment (azakus) [inline src/micro/extends.html]: Incomplete comment

Comment (sorvell) [inline src/micro/extends.html]: The name seems kinda general. Maybe `ElementConstructor` or something?

File: src/micro/extends.html [hunk 2]
```diff
@@ -87,6 +91,10 @@
       if (!p) {
         var np = this.getNativePrototype(tag);
         p = this.extend(Object.create(np), Polymer.Base);
+        var constructor = function(){};
```
Comment (sorvell) [inline src/micro/extends.html]: The function's not returning a prototype anymore. Let's change the name or add a new function?

File: src/micro/extends.html [hunk 3]
```diff
@@ -49,10 +49,14 @@
...
     _getExtendedPrototype: function(tag) {
       var isNative = tag && tag.indexOf('-') < 0;
-      var proto = isNative ? 
+      var proto = isNative ?
         this._getExtendedNativePrototype(tag) :
         this._getExtendedElementPrototype(tag);
-      proto.__extendsNativeElement = isNative;
+      // let `__extendsNativeElement` walk prototype chain to allow
+      // extensions of extensions of native elements
+      if (isNative) {
```
Comment (sorvell) [inline src/micro/extends.html]: Why this change?

File: src/micro/extends.html (hunk 2 above)
Comment (azakus) [inline src/micro/extends.html]: I'm not sure what you mean, the prototype is attached in the next line.

File: src/micro/extends.html (hunk 1 above)
Comment (azakus) [inline src/micro/extends.html]: Not a fan of the long function name
```js
class XFoo extends Polymer.ElementConstructor('other-element')
```

File: src/micro/extends.html (hunk 3 above)
Comment (azakus) [inline src/micro/extends.html]: If you set `__extendsNativeElement` on every prototype, then you can't extend elements that extend native elements to do the right call to `createElement`

```js
class XInput extends Polymer.Extend('input'){}
XInput.prototype.__extendsNativeElement === true;
// new XInput() -> <input is="x-input">
class XExtendInput extends XInput {}
XExtendInput.prototype.__extendsNativeElement === false;
// new XExtendInput() -> <x-extend-input>
```

**Comment by sorvell** [comment]
LGTM
